pkg/kubelet/hyper/hyperclient.go

```diff
@@ -788,24 +788,35 @@ func (client *HyperClient) IsImagePresent(repo, tag string) (bool, error) {
 }
 
 func (client *HyperClient) ListServices(podId string) ([]HyperService, error) {
-	v := url.Values{}
-	v.Set("podId", podId)
-	body, _, err := client.call("GET", "/service/list?"+v.Encode(), "", nil)
-	if err != nil {
-		if strings.Contains(err.Error(), "doesn't have services discovery") {
-			return nil, nil
-		} else {
-			return nil, err
-		}
+	request := grpctypes.ServiceListRequest{
+		PodID: podId,
 	}
 
-	var svcList []HyperService
-	err = json.Unmarshal(body, &svcList)
+	response, err := client.client.ServiceList(context.Background(), &request)
```

Comment:
You should also check whether the error is `doesn't have services discovery`or not.